platforms/Linux/centos/8/dev/swift-lang.spec

```diff
@@ -0,0 +1,170 @@
+Source10:       https://github.com/apple/indexstore-db/archive/swift-%{swifttag}.tar.gz#/indexstore-db.tar.gz
+Source11:       https://github.com/apple/llvm-project/archive/swift-%{swifttag}.tar.gz#/llvm-project.tar.gz
+Source12:       https://github.com/apple/swift-tools-support-core/archive/swift-%{swifttag}.tar.gz#/swift-tools-support-core.tar.gz
+Source13:       https://github.com/apple/swift-argument-parser/archive/%{sap_version}.tar.gz
+Source14:       https://github.com/apple/swift-driver/archive/swift-%{swifttag}.tar.gz#/swift-driver.tar.gz
+Source15:       https://github.com/unicode-org/icu/archive/release-%{icu_version}.tar.gz
+Source16:       https://github.com/apple/swift-syntax/archive/swift-%{swifttag}.zip#/swift-syntax.tar.gz
+Source17:       https://github.com/jpsim/Yams/archive/%{yams_version}.zip
+Source18:       https://github.com/apple/swift-crypto/archive/refs/tags/%{swift_crypto_version}.tar.gz
+
+Patch0:         patches/build-presets.patch
+Patch1:         patches/swift-api-checker.patch
+Patch2:         patches/hwasan_symbolize.patch
+
+BuildRequires:  autoconf
+BuildRequires:  clang
+BuildRequires:  cmake
+BuildRequires:  diffutils
+BuildRequires:  git
+BuildRequires:  glibc-static
+BuildRequires:  libbsd-devel
+BuildRequires:  libcurl-devel
+BuildRequires:  libedit-devel
+BuildRequires:  libicu-devel
+BuildRequires:  libstdc++-static
+BuildRequires:  libtool
+BuildRequires:  libuuid-devel
+BuildRequires:  libxml2-devel
+BuildRequires:  make
+BuildRequires:  ncurses-devel
+BuildRequires:  ninja-build
+BuildRequires:  pcre-devel
+BuildRequires:  python2
+BuildRequires:  python2-devel
+BuildRequires:  python2-six
+BuildRequires:  python3
+BuildRequires:  python3-six
+BuildRequires:  python3-pexpect
+BuildRequires:  platform-python-devel
+BuildRequires:  sqlite-devel
+BuildRequires:  swig
+BuildRequires:  rsync
+BuildRequires:  tar
+BuildRequires:  which
+
+Requires:       binutils
+Requires:       gcc
+Requires:       git
+Requires:       glibc-static
+Requires:       libbsd-devel
+Requires:       libedit
+Requires:       libedit-devel
+Requires:       libicu-devel
+Requires:       libstdc++-static
+Requires:       pkg-config
+Requires:       python3
+Requires:       sqlite
+Requires:       zlib-devel
+
+ExclusiveArch:  x86_64 aarch64
+
+%description
+Swift is a general-purpose programming language built using
+a modern approach to safety, performance, and software design
+patterns.
+
+The goal of the Swift project is to create the best available
+language for uses ranging from systems programming, to mobile
+and desktop apps, scaling up to cloud services. Most
+importantly, Swift is designed to make writing and maintaining
+correct programs easier for the developer.
+
+%prep
+# FIXME: what does -a 0 -a 1 etc do?
+%setup -q -c -n %{swiftbuild} -a 0 -a 1 -a 2 -a 3 -a 4 -a 5 -a 6 -a 7 -a 8 -a 9 -a 10 -a 11 -a 12 -a 13 -a 14 -a 15 -a 16 -a 17 -a 18
```

Comment:
The `-a` flag tells the `%setup` macro to expand the corresponding source tar ball after changing to the working directory - see https://rpm-packaging-guide.github.io/#setup

That's a good reference site for building RPMs in general.